harpy/_cli_types_generic.py

```
@@ -89,18 +90,19 @@ def convert(self, value, param, ctx):
-            self.fail(f"{value} is a file, but input should be a directory.", param, ctx)
-        if not os.path.exists(f"{value}/config.yaml"):
-            self.fail(f"{value} does not contain the necessary config.yaml file.", param, ctx)
-        elif not os.access(f"{value}/config.yaml", os.R_OK):
-            self.fail(f"{value}/config.yaml does not have read access. Please check the file permissions and try again.", param, ctx)
+        if os.path.isdir(value):
+            self.fail(f"{value} is a directory, but input should be a yaml file.", param, ctx)
+        if not os.access(value, os.R_OK):
+            self.fail(f"{value} is not readable. Please check file permissions and try again", param, ctx)
+        with open(value, "r") as file:
+            try:
+                data = yaml.safe_load(file)
```

Ruff (0.8.2)

Local variable `data` is assigned to but never used

Remove assignment to unused variable `data`

(F841)